manifests/exception.pp

```diff
@@ -69,26 +69,26 @@
 #   windows_firewall::exception { 'myapp':
 #     ensure       => present,
 #     direction    => 'in',
-#     action       => 'Allow',
+#     action       => 'allow',
 #     enabled      => 'yes',
 #     program      => 'C:\\myapp.exe',
 #     display_name => 'My App',
 #     description  => 'Inbound rule for My App',
 #   }
 #
 define windows_firewall::exception(
-  $ensure = 'present',
-  $direction = '',
-  $action = '',
-  $enabled = 'yes',
-  $protocol = '',
-  $local_port = '',
-  $remote_port = '',
+  Enum['present', 'absent'] $ensure = 'present',
+  Enum['in', 'out'] $direction = 'in',
+  Enum['allow', 'block'] $action = 'allow',
+  Variant[Enum['yes', 'no', 'true', 'false'], Boolean] $enabled = 'yes', # lint:ignore:quoted_booleans
```

Comment:
Given it'll be a major version bump anyway, why not take the opportunity to make it a true boolean?

Reply:
@ekohl I get your point. @alexjfisher What is your opinion here?

Reply:
Ah yes, if this is the first PR to break puppet 3, it will be a major release and I'm fine (and would encourage) switching to just Booleans.